tests/Unit.php

```diff
@@ -2,16 +2,86 @@
 
 class Unit extends PHPUnit_Framework_TestCase {
 
-	public function subject($params = array(), $options = array()) {
-		$options += array(
-			'mock' => false,
-			'methods' => array(),
+	public static $classNamespace = "alfmarks\\";
+
+	private $subjectClass;
+	private $subjectParams = array();
+	private $subjectOptions;
+
+	private $baseParams = array();
+
+	private $baseOptions = array(
+		'mock' => false,
+		'methods' => array()
+	);
+
+	public function __construct( ...$class ) {
+		if (!empty($class)) {
+			// Since we took in a splat
+			foreach($class as $klass) {
+				$this->subjectClass = $klass;
+			}
+		} else {
+			parent::__construct();
+		}
+	}
+
+	public static function getCalledClass() {
+		$calledClass = str_replace(
+			'Test', '', get_called_class()
 		);
-		$class = 'alfmarks\\' . str_replace('Test', '', get_called_class());
-		if ($options['mock']) {
-			return $this->getMock($class, $options['methods'], $params);
+		return self::$classNamespace . $calledClass;
+	}
+
+	public function withParams( ...$params ) {
```

Comment:
~~~php
	public function withParams( $params ) {
		$this->subjectParams = $params;
		return $this;
	}


->withParams(array(
  'key' => 'value',
  'foo' => 'bar',
));
~~~